services/checks/checks.go

```diff
@@ -94,32 +114,25 @@ func New(pmmVersion string) *Service {
 
 // Run runs checks service that grabs checks from Percona Checks service every interval until context is canceled.
 func (s *Service) Run(ctx context.Context) {
-	if f := os.Getenv(envCheckFile); f != "" {
-		s.l.Warnf("Using local test checks file: %s", f)
-
-		data, err := ioutil.ReadFile(f) //nolint:gosec
-		if err != nil {
-			s.l.Errorf("Failed to read test checks file: %s.", err)
-			return
-		}
-		checks, err := check.Parse(bytes.NewReader(data))
-		if err != nil {
-			s.l.Errorf("Failed to parse test checks file: %s.", err)
-			return
-		}
-
-		s.updateChecks(checks)
-		return
-	}
-
 	ticker := time.NewTicker(s.interval)
 	defer ticker.Stop()
 
+	go s.checkResults(ctx)
+
 	for {
-		if err := s.downloadChecks(ctx); err != nil {
-			s.l.Errorf("Failed to download checks: %v", err)
+		if f := os.Getenv(envCheckFile); f != "" {
+			s.l.Warnf("Use local test checks file: %s.", f)
+			if err := s.loadLocalChecks(f); err != nil {
+				s.l.Errorf("Failed to load local checks file: %s.", err)
+			}
+		} else {
+			if err := s.downloadChecks(ctx); err != nil {
+				s.l.Errorf("Failed to download checks, %s.", err)
+			}
 		}
```

Comment:
I would extract this block to separate function named something like `updateChecks`

Reply:
Done, extracted to `grabChecks` function, as `updateChecks` func already exists and updatates slice filed where checks stored.